client/components/domains/domain-product-price/index.jsx

```diff
@@ -1,74 +1,81 @@
-		} else if ( this.shouldShowPremiumMessage() ) {
-			return (
+import PremiumPopover from 'components/plans/premium-popover';
+const DomainProductPrice = React.createClass( {
+	propTypes: {
+		isLoading: React.PropTypes.bool,
+		price: React.PropTypes.string,
+		freeWithPlan: React.PropTypes.bool,
+		requiresPlan: React.PropTypes.bool
+	},
+	renderFreeWithPlan() {
+		return (
+			<div className="domain-product-price is-free-domain">
+				<span className="domain-product-price__price">{ this.translate( '%(cost)s {{small}}/year{{/small}}', {
+					args: { cost: this.props.price },
+					components: { small: <small /> }
+				} ) }</span>
+				<span className="domain-product-price__free-text" ref="subMessage">
+					{ this.translate( 'Free with your plan' ) }
+				</span>
+			</div>
+		);
+	},
+	renderFree() {
+		return (
+			<div className="domain-product-price">
+				<span className="domain-product-price__price">{ this.translate( 'Free' ) }</span>
+			</div>
+		);
+	},
+	renderIncludedInPremium() {
+		return (
+			<div className="domain-product-price is-with-plans-only">
 				<small className="domain-product-price__premium-text" ref="subMessage">
 					{ this.translate( 'Included in WordPress.com Premium' ) }
 					<PremiumPopover
 						context={ this.refs && this.refs.subMessage }
 						bindContextEvents
 						position="bottom left"/>
 				</small>
-			);
-		}
-		return null;
-	},
-	priceMessage( price ) {
-		return this.translate( '%(cost)s {{small}}/year{{/small}}', {
-			args: { cost: price },
-			components: { small: <small /> }
-		} );
+			</div>
+		);
 	},
-	priceText() {
-		const selectedSite = sitesList.getSelectedSite();
-		if ( ! this.props.price ) {
-			return this.translate( 'Free' );
-		} else if ( domainsWithPlansOnlyTestEnabled && ! cartItems.isNextDomainFree( this.props.cart ) && ( ! selectedSite || ! isPlan( selectedSite.plan ) ) ) {
-			return null;
-		}
-		return this.priceMessage( this.props.price );
+	renderPrice() {
+		return (
+			<div className="domain-product-price">
+				<span className="domain-product-price__price">
+					{ this.translate( '%(cost)s {{small}}/year{{/small}}', {
+						args: { cost: this.props.price },
+						components: { small: <small /> }
+					} ) }
+				</span>
+			</div>
+		);
 	},
-	render: function() {
-		const classes = classNames( 'domain-product-price', {
-				'is-free-domain': cartItems.isNextDomainFree( this.props.cart ),
-				'is-with-plans-only': this.shouldShowPremiumMessage(),
-				'is-placeholder': this.props.isLoading
-			} );
-
+	render() {
 		if ( this.props.isLoading ) {
-			return <div className={ classes }>{ this.translate( 'Loading…' ) }</div>;
+			return <div className="domain-product-price is-placeholder">{ this.translate( 'Loading…' ) }</div>;
 		}
 
-		return (
-			<div className={ classes }>
-				<span className="domain-product-price__price">{ this.priceText() }</span>
-				{ this.subMessage() }
-			</div>
-		);
+		switch ( this.props.rule ) {
```

Comment:
I understand you wanted to simplify this method (especially getting rid of the complicated `subMessage` method), but it seems that the render* methods you've introduced are very similar to each other. Not sure if it's worth the extra effort to extract a common render method for them (maybe with the exception of the Premium case?), but it just seems excessive to have basically the same `div`s in three separate methods.